src/gobapi/storage.py

```diff
@@ -98,26 +117,75 @@ def get_entities(collection_name, offset, limit):
     return (entities, all_count)
 
 
-def get_entity(collection_name, id):
+def _get_entities_from_view(view_name, offset, limit):
+    assert(session and Base)
+    view = Table(view_name, metadata, autoload=True)
+    all_entities = session.query(view)
+
+    all_count = all_entities.count()
+
+    metadata_column_list = [k for k in {**PUBLIC_META_FIELDS, **PRIVATE_META_FIELDS, **FIXED_COLUMNS}.keys()]
+    columns = [c for c in view.columns if c.name not in metadata_column_list]
+
+    entities = [
+        _entity_to_dict_from_sql_type(entity, columns)
+        for entity in all_entities.offset(offset).limit(limit).all()
+    ]
+
+    return (entities, all_count)
+
+
+def get_entity(collection_name, id, view=None):
     """Entity
 
     Returns the entity within the specified collection identied by the id parameter.
     If the entity cannot be found, None is returned
 
     :param collection_name:
     :param id:
+    :param view:
     :return:
     """
     assert(session and Base)
-    table, model = _get_table_and_model(collection_name)
+    if view:
+        return _get_entity_from_view(view, id)
+    else:
+        return _get_entity_from_collection(collection_name, id)
+
 
-    id_field = model['entity_id']
+def _get_entity_from_collection(collection_name, id):
+    table, model = _get_table_and_model(collection_name)
 
     filter = {
-        id_field: id,
+        "_id": id,
         "_date_deleted": None
     }
 
     entity = session.query(table).filter_by(**filter).one_or_none()
 
     return _entity_to_dict(entity, model, PUBLIC_META_FIELDS) if entity else None
+
+
+def _get_entity_from_view(view_name, id):
+    """Entity
+
+    Returns the entity within the specified view identied by the id parameter.
+    If the entity cannot be found, None is returned
+
+    :param view_name:
+    :param id:
+    :return:
+    """
+    assert(session and Base)
+    view = Table(view_name, metadata, autoload=True)
+
+    filter = {
+        "_id": id,
+        "_date_deleted": None
+    }
```

Comment:
I suggest having this filter only once, now it is duplicated in both get_entitiy's